### ISSUE TITLE
Set up structure and process for localization

### ISSUE THREAD

**Issue by crandmck** [loopbackio/loopback.io#33]
Need to establish a framework for localization.  This includes:
- Basic internationalization infrastructure along the lines of what we have at http://expressjs.com.
- Copying existing translations listed in https://docs.strongloop.com/display/public/LB/Docs+in+other+languages (as appropriate)
- Writing up procedure for translators to follow.

**Comment by crandmck** [comment]
Based on how we do it in expressjs.com, we'll need to create folders for each language under the `pages` folder, for example `en` for English, `de` for German, `zh` for Chinese, etc.
These folders would then contain the docs in that language, e.g. `lb2` and `lb3` folders, for LoopBack 2.x and LoopBack 3.  Some languages might decide to just have 3.0 content or 2.x content.

This will also affect how we write links: we have to add a `lang` property to the front-matter and `{{page.lang}}` to links.

**Comment by nuba** [comment]
Here's a thread describing and linking to a number of different approaches for making multilingual sites with jekyll https://talk.jekyllrb.com/t/multi-language-strategies-with-jekyll/315

**Comment by crandmck** [comment]
I think this is the best approach since it will work with GH Pages: https://www.sylvaindurand.org/making-jekyll-multilingual/  I like how it provides a language selector on each page.

It's similar to what we use on http://expressjs.com/.  I need to look into it a bit more to see how we would implement this with our Jekyll template; specifically, how we would add this to the layout templates that we have.

**Comment by crandmck** [comment]
I added a "Translations" dropdown selector in the top bar, and added some languages as examples.  I documented the basic structure and process in:
- http://loopback.io/doc/en/contrib/Adding_a_new_language.html
- http://loopback.io/doc/en/contrib/Translating_articles.html

I didn't add anything fancy (yet) to link to the other languages; We could make a number of improvements, but the basic setup is there to enable translation, so I'm going to close this issue.  Let's add separate issues for any further improvements, or if there are problems with the current implementation.
